_posts/2022-05-27-infinispan-14.0.0.Dev03.adoc

```diff
@@ -0,0 +1,101 @@
+---
+layout: blog
+title: Infinispan 14.0.0.Dev03
+permalink: /blog/:year/:month/:day/infinispan-14
+date: '2022-05-27T12:00:00.000-00:00'
+author: ttarrant
+tags: [ "release", "development" ]
+---
+
+= Infinispan 14.0.0.Dev03
+
+Dear Infinispan community,
+
+Infinispan 14 development release 03 is here! We plan to release Infinispan 14 Final this summer, but we want to share a preview of what's coming.
+
+== Java 11
+
+Infinispan now requires Java 11 to run (it was only needed to build it before). This means we can use and expose all of the great new APIs that were added, such as `java.util.concurrent.Flow` which provides a standard interface for all things reactive.
+Which brings us to the...
+
+== New API
+
+We have finalized the design of our new user-facing API, which brings the following, much-needed, features:
+
+* a common API for both embedded and remote
+* clean separation between sync and async APIs, as well as a link:https://smallrye.io/smallrye-mutiny/[Mutiny] variant which blends beautifully with all the great things happening over in link:https://quarkus.io/[Quarkus].
+* a single entry-point to access all of the data-structures that we support (caches, counters, locks, multimaps, etc)
+* our own annotations for indexing entity fields (see below for details)
+
+We are now working on implementing this API for the remote Hot Rod client, while the implementation for embedded will be done in Infinispan 15.
+
+
+== JGroups 5
+
+Upgrading to Java 11 also allows us to upgrade to JGroups 5.x, which brings a bunch of improvements:
+
+* Improved failure-detection protocols (`FD_ALL3`, `FD_SOCK2`)
```

Comment:
we are not using FD_SOCK2 yet. I went through the release notes and I found these new protocols. Created [1] to update the stacks in the next release

FD_SOCK2
VERIFY_SUSPECT2
FRAG4

[1] https://issues.redhat.com/browse/ISPN-13929